dbt/adapters/impala/connections.py

```diff
@@ -248,12 +248,32 @@ def add_query(
             self.begin()
         fire_event(ConnectionUsed(conn_type=self.TYPE, conn_name=connection.name))
 
+        additional_info = {}
+        if self.query_header:
+            try:
+                additional_info = json.loads(self.query_header.comment.query_comment.strip())
+            except Exception:  # silently ignore error for parsing
+                additional_info = {}
```

Comment:
add a debug log to indicate that we failed getting query header because we are swallowing this exception.